gapis/api/state.go

```diff
@@ -142,15 +150,24 @@ func (s GlobalState) MemoryDecoder(ctx context.Context, d memory.Data) *memory.D
 
 // MemoryEncoder returns a memory encoder using the state's memory layout
 // to encode to the pool p, for the range rng.
-func (s GlobalState) MemoryEncoder(p memory.PoolID, rng memory.Range) *memory.Encoder {
-	return memory.NewEncoder(s.MemoryWriter(p, rng), s.MemoryLayout)
+func (s GlobalState) MemoryEncoder(ctx context.Context, p memory.PoolID, rng memory.Range) *memory.Encoder {
+	return memory.NewEncoder(s.MemoryWriter(ctx, p, rng), s.MemoryLayout)
 }
 
 // Alloc allocates a memory range using the Allocator associated with
 // the given State, and returns a AllocResult that can be used to access the
 // pointer, and range.
 func (s *GlobalState) Alloc(ctx context.Context, size uint64) (AllocResult, error) {
-	at, err := s.Allocator.Alloc(size, 8)
+	// Force at least one byte to be allocated,
```

Comment:
We could also just return nullptr here, and handle frees of nullptr specially. (a-la malloc/free)